tests/test_ext_todo.py

```diff
@@ -84,3 +84,31 @@ def on_todo_defined(app, node):
     # check handled event
     assert len(todos) == 2
     assert set(todo[1].astext() for todo in todos) == set(['todo in foo', 'todo in bar'])
+
+@pytest.mark.sphinx('latex', testroot='ext-todo', freshenv=True,
+                    confoverrides={'todo_include_todos': True, 'todo_emit_warnings': True})
+def test_todo_valid_link(app, status, warning):
+    """
+    Test that the inserted "original entry" links for todo items have a target
+    that exists in the LaTeX output. The target was previously incorrectly
+    omitted (GitHub issue #1020).
+    """
+
+    # Ensure the LaTeX output is built.
+    app.builder.build_all()
+
+    content = (app.outdir / 'TodoTests.tex').text()
+
+    # Look for the link to foo. We could equally well look for the link to bar.
+    link = r'\{\\hyperref\[\\detokenize\{(.*?foo.*?)}]\{\\sphinxcrossref{' \
+        r'\\sphinxstyleemphasis{original entry}}}}'
+    m = re.findall(link, content)
+    assert len(m) == 1
+    target = m[0]
+
+    # Look for the targets of this link.
+    labels = [m for m in re.findall(r'\label{([^}]*)}', content)
```

Comment:
It seems this regexp causes errors on Travis CI.

Reply:
Naughty me didn't test on Python 3.6... It seems unknown escapes became an error after 3.5. I don't have an accessible 3.6 right now but I've pushed an optimistic attempt. If it doesn't pass I'll debug it when my 3.6 is reachable again. Apologies for the noise.

Reply:
Don't mind. We use Travis CI for this purpose :-)